### PR TITLE
MM-37632: Custom post for status updates

### PR DESCRIPTION
**Pull request by agarciamontoro**
#### Summary

This PR adds a custom post for rendering status updates in both the run channel and the broadcast channels. The information shown in the footer (number of tasks and number of participants) needed in the post is passed as props in the post object from the server. The finishing message was also modified to match the <author> <action> <object> structure of all other messages.

You can review by commit if you want to :)

##### Status updates in the run channel

image

##### Themed variant

image

##### Detail of participants tooltip

image

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-37632

#### Checklist

- [ ] ~~Telemetry updated~~
- [ ] ~~Gated by experimental feature flag~~
- [X] Unit tests updated

### CODE REVIEW

File: webapp/src/components/update_post.tsx [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+        singleline: false,
+        mentionHighlight: true,
+        atMentions: true,
+        team,
+        channelNamesMap,
+    };
+
+    const mdText = (text: string) => messageHtmlToComponent(formatText(text, markdownOptions), true, {});
+
+    const numTasksChecked = props.post.props.numTasksChecked ?? 0;
```
Comment (crspeller) [inline webapp/src/components/update_post.tsx:43]: TIL `??`

**Review by crspeller — APPROVED** [review]

File: webapp/src/components/update_post.tsx [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+import {messageHtmlToComponent, formatText} from 'src/webapp_globals';
+import {ChannelNamesMap} from 'src/types/backstage';
+import {useFormattedUsernameByID} from 'src/hooks/general';
+import {currentPlaybookRun} from 'src/selectors';
+
+interface Props {
+    post: Post;
+}
+
+export const UpdatePost = (props: Props) => {
```
Comment (cpoile) [inline webapp/src/components/update_post.tsx:27]: nice component, clean. 👍

**Review by cpoile — APPROVED** [review]
Nice!

**Comment by agarciamontoro** [comment]
I guess these are the E2E tests that are failing everywhere, right?

**Comment by agarciamontoro** [comment]
Merging as the E2E tests are being fixed in webapp by https://github.com/mattermost/mattermost-webapp/pull/8886

**Pull request merged**